client/blocks/product-plan-overlap-notices/index.jsx

```diff
@@ -95,6 +96,12 @@ class ProductPlanOverlapNotices extends Component {
 		return availableProducts[ currentPlanSlug ].product_name;
 	}
 
+	clickPurchaseHandler = purchaseId => {
+		this.props.recordTracksEvent( 'calypso_product_overlap_purchase_view', {
```

Comment:
Inseatead of `calypso_product_overlap_purchase_view` I would call it `calypso_product_overlap_purchase_click` 
Since it is the click event that we are tracking here. 

```suggestion
		this.props.recordTracksEvent( 'calypso_product_overlap_purchase_click', {
```